App/Talent.App.WebApp/wwwroot/js/react/ReactScripts/Profile/Experience.jsx

```diff
@@ -1,15 +1,759 @@
+        
+
+        this.setState({
+            newExp: data,
+
+        })
+
+        
+    }
+
+    handleDateChange(event, { name, value }) {
+        if (this.state.hasOwnProperty(name)) {
+            this.setState({ [name]: value });
+        }
+        
+    }
+
+            
+        
+   handleChangeDate(date, name) {
+        var data = Object.assign({}, this.state.newExp);
+        
+       data[name] = date;
+        this.setState({
+            newExp: data,
+        })
+        
+    }
+
+    
+
+    addRecord() {
+
+        const l = this.props.experienceData;
+                       
+        if (this.state.newExp.company == "" || this.state.newExp.position == "") {
+            alert("company and position cannot be empty");
+            return;
+        }
+
+        if (this.state.newExp.start > this.state.newExp.end) {
+            alert("End date must be later than start end");
+            return;
+        }
+
+        var h = this.state.newExp;
+        
+        var cookies = Cookies.get('talentAuthToken');
+        $.ajax({
+            url: 'https://mvpstandard-p.azurewebsites.net/profile/profile/addExperience',
+            
+            headers: {
+                'Authorization': 'Bearer ' + cookies,
+                'Content-Type': 'application/json'
+            },
+            type: "POST",
+            data: JSON.stringify(h),
+            success: function (res) {
+                
+                if (res.success == true) {
+                    TalentUtil.notification.show("Experience added sucessfully", "success", null, null)
+                    this.loadData().then(() => { // Load data after state has been updated
+                        this.props.updateProfileData(this.props.componentId, this.state.experienceData);
+                    });                                         
+                                        
+
+                } else {
+                    TalentUtil.notification.show("Experience did not update successfully", "error", null, null)
+                }
+
+            }.bind(this),
+            error: function (res, a, b) {
+                console.log(res)
+                console.log(a)
+                console.log(b)
+            }
+        });              
+
+        this.setState({
+            newExp: {
+                company: "",
+                position: "",
+                responsibilities: "",
+                start: null,
+                end: null
+            },
+        });
+
+    }
+
+
+    updateExperience(data) {        
+        
+        const l = this.state.experienceData;
+        if (data.company == "" || data.position == "") {
+            alert("pls fill update");
+            return;
+        }
+
+        if (data.start > data.end) {
+            alert("End date must be later than start end");
+            return;
+        } 
+
+        
+        var result = l.findIndex(item => item.id == data.id)
+        
+        l[result] = data;
```

Comment:
What is **l**? Please avoid this variable convention and be more specific about it. Please be mindful that other developers won't be able to understand easily what are these variables.